django-s3-file-field-client/src/client.ts

```diff
@@ -25,11 +31,51 @@ export interface UploadResult {
   state: 'aborted' | 'successful' | 'error';
 }
 
-export default class S3FFClient {
-  protected baseUrl: string;
+class Progressable {
+  private overheadPercentage = 0.0;
+  private mainPercentageSegments: number[] = [];
+  // This value was picked arbitrarily
+  private readonly overheadProportion = 0.20;
```

Comment:
I'm not fond of this statically fixed overhead proportion. If we insist on not using indeterminate progress, let's at least base the proportion intelligently on the file's size.